### PR TITLE
2.3.0 Release

### PR DESCRIPTION
**Pull request by nblumhardt**
* #30, #34, #39, #40 - improve factoring, allow implementations to override "eager-first-event" behavior (@kimbirkelund, @oliverholliday, @nblumhardt)

### CODE REVIEW

File: src/Serilog.Sinks.PeriodicBatching/Sinks/PeriodicBatching/BoundedConcurrentQueue.cs [hunk 1]
```diff
@@ -1,36 +1,45 @@
...
 using System.Collections.Concurrent;
 using System.Threading;
 
 namespace Serilog.Sinks.PeriodicBatching
 {
-    class BoundedConcurrentQueue<T> 
+    class BoundedConcurrentQueue<T>
     {
-        const int NON_BOUNDED = -1;
+        public const int NonBounded = -1;
```
Comment (julealgon) [inline src/Serilog.Sinks.PeriodicBatching/Sinks/PeriodicBatching/BoundedConcurrentQueue.cs]: Are we tracking to rename this to `Unbounded` on the next release? `NonBounded` is not proper English.

**Review by julealgon — APPROVED** [review]

**Comment by julealgon** [comment]
Eagerly awaiting for this one! Do we have an ETA? I assume the package will be published as soon as this is merged?

File: src/Serilog.Sinks.PeriodicBatching/Sinks/PeriodicBatching/BoundedConcurrentQueue.cs (hunk 1 above)
Comment (kimbirkelund) [reply src/Serilog.Sinks.PeriodicBatching/Sinks/PeriodicBatching/BoundedConcurrentQueue.cs]: Can't believe I didn't catch that when writing it.

Comment (nblumhardt) [reply src/Serilog.Sinks.PeriodicBatching/Sinks/PeriodicBatching/BoundedConcurrentQueue.cs]: :-) I fixed the XML parameter docs but forgot about this; I'll send one though (though it's not public API surface)

**Comment by nblumhardt** [comment]
Hi @julealgon - yes, this will be published as soon as it's merged. Ideally we'd get some more feedback on the new APIs before setting them in stone, but if we don't hear any more this week I'll hit the button 👍

**Pull request merged**